lib/posix-socket/driver_list.ld

```diff
@@ -0,0 +1,10 @@
+SECTIONS
+{
+	. = ALIGN(0x8);
```

Comment:
> > Hi @craciunoiuc, @skuenzer,
> > To avoid misunderstanding, I'd like to add that this library provides (at least at the moment) only the driver API and the socket syscalls. So `posix-socket` does not actually implement the socket API but only functions as the POSIX socket interface to applications and other libraries. It requires `lwip` to register a driver for `AF_INET` and throw out its entire syscall interface. However, we still depend on `lwip`'s socket implementation (except `epoll`/`select`).
> > I have internally a modified `lwip` that connects to `posix-socket` but I have not yet created a PR. I doubt it will make it in the release tomorrow.
> 
> Yes, that's what I thought. Maybe then we should add somewhere in the config a mention that it's at-the-moment not usable with lwip?
> 
> Or should we just move fast and integrate that too into lwip?

Let me see if I can make a PR for it as the code should be mostly ok.

Reply:
I have not had any problems with it yet and I tested with a lot of other libraries and also on KVM and XEN. If you remove the alignment the driver list will be put at a byte-aligned location which is not good for performance and can cause problems if the architecture does not support unaligned memory accesses. So I would keep it for now.

Reply:
I didn't have problems with `ALIGN` either when I wanted to use it. After that bug I removed it as per @skuenzer 's requests.

@skuenzer should `ALIGN` stay here or not? For me it's fine either way